### PR TITLE
update config for 24.1

### PR DESCRIPTION
**Pull request by bgruening**

### CODE REVIEW

File: group_vars/gxconfig.yml [hunk 1]
```diff
@@ -1130,7 +1185,7 @@ base_app_main: &BASE_APP_MAIN
   # via a proxy server.  These options should be self explanatory and so
   # are not documented individually.  You can use these paths (or ones
   # in the proxy server) to point to your own styles.
-  static_enabled: true
+  #static_enabled: true
```
Comment (bgruening) [inline group_vars/gxconfig.yml:1188]: it should be the default value

File: files/galaxy/config/file_source_templates.yml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+- include: ./lib/galaxy/files/templates/examples/production_azure.yml
```
Comment (bgruening) [inline files/galaxy/config/file_source_templates.yml:1]: we need to decide what we do with those templates ...
Do we copy them into a specific location? How do we handle version updates to those files then? I don't think we need to template them out, but we need to decide where to store them.

Comment (bgruening) [reply files/galaxy/config/file_source_templates.yml:1]: a few thoughts:

* we might want to add new templates, and those are then most likely not in the Galaxy source tree
* upstream might add updates - a new version of the file-source with a new version number
  *  those we could sync like we sync now other config files
* I assume those templates need to be accessible by celery and other services - so on a shared filesystem

Maybe a separate folder under mutable-config/file-source-templates and mutable-config/object-store-templates/ ?

Comment (bgruening) [reply files/galaxy/config/file_source_templates.yml:1]: @jmchilton any thoughts?

File: group_vars/gxconfig.yml (hunk 1 above)
Comment (sanjaysrikakulam) [reply group_vars/gxconfig.yml:1188]: It is the [default value](https://github.com/galaxyproject/galaxy/blob/release_24.1/lib/galaxy/config/schemas/config_schema.yml#L1768)

File: files/galaxy/config/file_source_templates.yml (hunk 2 above)
Comment (sanjaysrikakulam) [reply files/galaxy/config/file_source_templates.yml:1]: I think that is a good solution. We can have a separate directory and append it to the lists [here](https://github.com/usegalaxy-eu/infrastructure-playbook/blob/master/group_vars/sn06.yml#L378-L469). We can add a couple of variables there and let the `ansible-galaxy` role copy everything and handle these files.

**Comment by sanjaysrikakulam** [comment]
I have added a new commit that:
1. Adds object store and file source templates and the catalog for the same.
2. Updates galaxy upgrade diff script to include the `compare` for these templates.
3. Enables the file source and object store template variables in the galaxy conf.
4. sn06 group vars are updated that allow the creation of the new dirs `file_source_templates` and `object_store_templates` under `galaxy_mutable_config_dir.` Also, copies the file source and object store templates along with the catalog files to the respective locations on the server. 

I deployed the same in the [ESG instance](https://github.com/sanjaysrikakulam/esg_wp4_galaxy/pull/4), and the object store and file source options are now available for the users in the user preferences.

File: bin/diff-before-update [hunk 3]
```diff
@@ -51,3 +51,19 @@ compare files/$folder/config/user_preferences_extra_conf.yml	$GALAXY/config/user
 #compare templates/$folder/config/build_sites.yml.j2	$GALAXY/config/build_sites.yml.sample.sample
 
 compare group_vars/gxconfig${ext}.yml	$GALAXY/config/galaxy.yml.sample
+
+# Compare file source templates
+compare files/$folder/config/file_source_templates/production_aws_private_bucket.yml	$GALAXY/files/templates/examples/production_aws_private_bucket.yml
```
Comment (bgruening) [inline bin/diff-before-update:56]: Any idea how we remember to add new templates here if we add new ones?

Comment (sanjaysrikakulam) [reply bin/diff-before-update:56]: I will update our operations [here](https://github.com/usegalaxy-eu/operations/blob/main/upgrade.md) and add additional checkpoints there that this script should updated if and when new templates from upstream are added.

Comment (sanjaysrikakulam) [reply bin/diff-before-update:56]: Additionally, we can loop over all the files in our directory and compare them to the files in the respective templates dir in the upstream and compare them during the upgrade cycle.

Comment (mira-miracoli) [reply bin/diff-before-update:56]: Could this be done with a github action that checks what upsteam has and compares it to master and then fails if not in sync?

Comment (sanjaysrikakulam) [reply bin/diff-before-update:56]: unfortunately not because the upstream has more templates; we picked only the ones that are "production" ready.

**Comment by sanjaysrikakulam** [comment]
BTW: I created an issue in the upstream related to the object store https://github.com/galaxyproject/galaxy/issues/18500

We either have to wait a bit or proceed with the rollout without the object stores and deploy object stores later.

**Comment by bgruening** [comment]
We can disable this feature for the time being.

**Comment by bgruening** [comment]
A new NFS is ordered and we will use that when we advertise this feature broadly.

**Comment by bgruening** [comment]
So lets disable this feature then.

**Comment by sanjaysrikakulam** [comment]
I have now disabled copying the object store and file source template catalogs. 

I think the PR is ready for review. Please check.

File: bin/diff-before-update (hunk 3 above)
Comment (sanjaysrikakulam) [reply bin/diff-before-update:56]: Documentation [PR](https://github.com/usegalaxy-eu/operations/pull/46)

**Comment by bgruening** [comment]
Good for me. I'm I allowed to do the update this evening or better tomorrow evening?

**Comment by sanjaysrikakulam** [comment]
Tomorrow evening is better. 

By the way, this [patch] (https://github.com/galaxyproject/galaxy/pull/18520) fixes the object store [issue] (https://github.com/galaxyproject/galaxy/issues/18500). When we have the new NFS box, we can deploy the user file source and the object store features.

**Comment by bgruening** [comment]
So any last comments before I deploy it this evening?

**Review by sanjaysrikakulam — APPROVED** [review]

**Review by mira-miracoli — APPROVED** [review]

**Pull request merged**